Update mutation-testing.md (#46064)
dotnet/docs: docs/core/testing/mutation-testing.md

For additional mutation types, see the [Stryker.NET: Mutations](https://stryker-mutator.io/docs/stryker-net/mutations) documentation.

## Interpreting Mutation Testing Results

After running Stryker.NET, you’ll receive a report that categorizes mutants as **killed**, **survived**, or **timeout**. Here's how to interpret and act on these results:

- **Killed Mutants**: These are changes that your tests successfully caught. A high number of killed mutants indicates that your test suite effectively detects logic errors.

- **Survived Mutants**: These changes were not caught by your tests. Review them to identify gaps in test coverage or assertions that are too weak. Focus on adding targeted unit tests that would fail if the mutant were real.

- **Timeout Mutants**: These mutations caused your code to hang or exceed the allowed time. This can happen with infinite loops or unoptimized paths. Investigate the code logic or increase the timeout threshold if needed.

> **Note**: Don't chase a 100% mutation score. Focus instead on high risk or business critical areas where undetected bugs would be most costly.

> ## CI/CD Pipelines
> Mutation testing can be easily integrated into your CI/CD pipelines. For example, you can configure Stryker.NET to run as part of your Azure Pipelines (or GitHub Actions) workflow and enforce quality gates based on thresholds.

Reviewer: ```suggestion
## Adding mutation testing to your CI/CD workflow

You can seamlessly integrate mutation testing into your continuous integration and delivery workflows. For instance, Stryker.NET can be configured to run within your Azure Pipelines or GitHub Actions setup, allowing you to enforce quality thresholds as part of your automated testing process.
```

```
  "thresholds": {
    "high": 85,
    "low": 65,
    "break": 0
  }

> ```

Author: We can remove it, but we are talking about "quality thresholds". This is an example how a threshold looks like. It's your call if you want it gone or not.

> ## Costumization
> Besides setting thresholds for your pipeline, Stryker.NET offers the possibility of having different configurations for each of your project needs. You can do this customisation of behaviour using the stryker-config.json file.

Reviewer: ```suggestion

## Customization

Besides setting thresholds for your pipeline, Stryker.NET offers the possibility of having different configurations for each of your project needs. You can do this customization of behavior using the _stryker-config.json_ file.
```

> ```

Reviewer: ```suggestion
```json
```

{

>   "ignoreMutations": [

Reviewer: Why is the naming convention of the config keys different? In one case we see `ignoreMutations` and in the other we see `ignore-methods`? Are both acceptable?

Author: My mistake. Ignore-mutations is the way to go based on their doc

>   "ignoreMutations": [

Reviewer: ```suggestion
  "ignore-mutations": [
```

    "ToString",
    "Equals",
    "GetHashCode"
  ],
 "ignore-methods": [
        "*Logs"
    ]
  "mutate": [
    "!**/Migrations/*",
    "!**/*.Designer.cs"
  ]
}
```

> - **IgnoreMutations**: These are the methods or expressions that you want to ignore because they are noisy or you consider not needed based on your application logic needs. They will show up in your reports as Ignored.
> - **Ignore-methods**: You can use this to skip entire methods based on their signatures. Also, those will appear in your reports as Ignored. In our example, we ignore all methods ending in "Logs".
> - **Mutate**: Without this option, Styker will try to mutate all the files in your project. With this, you can ignore files or entire folders. In the example above, we ignore everything inside a Migrations folder and all .Designer.cs files (which are usually autogenerated) 

Reviewer: ```suggestion

- **ignore-mutations**: These are the methods or expressions that you want to ignore because they are noisy, or you consider not needed based on your application logic needs. They will show up in your reports as `Ignored`.
- **ignore-methods**: You can use this to skip entire methods based on their signatures. Also, those will appear in your reports as Ignored. In our example, we ignore all methods ending in "Logs".
- **mutate**: Without this option, Styker will try to mutate all the files in your project. With this, you can ignore files or entire folders. In the example above, we ignore everything inside a _Migrations_ folder and all _.Designer.cs_ files (which are usually autogenerated) .
```

> 

Reviewer: ```suggestion

For more information, see [Stryker: Configuration](https://stryker-mutator.io/docs/stryker-net/configuration/).

```
